Lib/csharp/arrays_csharp.i

```diff
@@ -94,13 +94,15 @@ CSHARP_ARRAYS(short, short)
 CSHARP_ARRAYS(unsigned short, ushort)
 CSHARP_ARRAYS(int, int)
 CSHARP_ARRAYS(unsigned int, uint)
-// FIXME - on Unix 64 bit, long is 8 bytes but is 4 bytes on Windows 64 bit.
-//         How can this be handled sensibly?
-//         See e.g. http://www.xml.com/ldd/chapter/book/ch10.html
+#if defined(SWIGWORDSIZE64)
+CSHARP_ARRAYS(long, long)
+CSHARP_ARRAYS(unsigned long, ulong)
+#else
 CSHARP_ARRAYS(long, int)
 CSHARP_ARRAYS(unsigned long, uint)
 CSHARP_ARRAYS(long long, long)
 CSHARP_ARRAYS(unsigned long long, ulong)
```

Comment:
I'm not sure if it's correct to exclude typemaps for `[unsigned] long long` completely when `SWIGWORDSIZE64` is defined. Of course, it has the same size as `long` in this case, but we should still have typemaps for it, shouldn't we?

Reply:
+1
note: In 32bits we already have a "map collision" between `int` and `long` (mapped to .Net `int`).
So in 64bits having `long` and `long long` mapped to .Net `long` look good to me.

Reply:
Are we sure long long is 64 bits wide? I was only using stdint.h so I didn't worry about that type.

Reply:
According to my tests on 64bits OS `sizeof(long long) == 8`
ref: https://github.com/google/or-tools/blob/stable/cmake/doc/swig.md